cincinnati/src/plugins/internal/arch_filter.rs

```diff
@@ -0,0 +1,343 @@
+            ),
+            (
+                1,
+                [
+                    (String::from("version_suffix"), String::from("+amd64")),
+                    (String::from("release.arch"), String::from("amd64")),
+                ]
+                .iter()
+                .cloned()
+                .collect(),
+            ),
+            (
+                2,
+                [
+                    (String::from("version_suffix"), String::from("+arm64")),
+                    (String::from("release.arch"), String::from("arm64")),
+                ]
+                .iter()
+                .cloned()
+                .collect(),
+            ),
+            (
+                3,
+                [
+                    (String::from("version_suffix"), String::from("+arm64")),
+                    (String::from("release.arch"), String::from("arm64")),
+                ]
+                .iter()
+                .cloned()
+                .collect(),
+            ),
+        ];
+        let input_edges = Some(vec![(0, 1), (2, 3)]);
+        let input_graph: cincinnati::Graph =
+            generate_custom_graph("image", input_metadata.clone(), input_edges.to_owned());
+
+        // filter by arm64
+        let expected_metadata: TestMetadata = vec![
+            (
+                2,
+                [(String::from("release.arch"), String::from("arm64"))]
+                    .iter()
+                    .cloned()
+                    .collect(),
+            ),
+            (
+                3,
+                [(String::from("release.arch"), String::from("arm64"))]
+                    .iter()
+                    .cloned()
+                    .collect(),
+            ),
+        ];
+        let expected_edges = Some(vec![(0, 1)]);
+
+        let expected_graph: cincinnati::Graph =
+            generate_custom_graph("image", expected_metadata, expected_edges.to_owned());
+
+        let future_processed_graph = Box::new(ArchFilterPlugin {
+            key_prefix: "release".to_string(),
+            key_suffix: "arch".to_string(),
+            default_arch: "amd64".to_string(),
+            default_arch_threshold_version: "1.0.0".to_string(),
+        })
+        .run_internal(InternalIO {
+            graph: input_graph.clone(),
+            parameters: [("arch", "arm64")]
+                .iter()
+                .map(|(k, v)| (k.to_string(), v.to_string()))
+                .collect(),
+        });
+
+        let processed_graph = runtime.block_on(future_processed_graph)?.graph;
+
+        assert_eq!(expected_graph, processed_graph);
+
+        Ok(())
+    }
+
+    #[test]
+    fn ensure_infer_version() -> Fallible<()> {
+        // (arch, version, default_arch, default_arch_threshold), expecteded_arch
+        [
+            ((None, Some("1.0.0"), "amd64", "2.0.0"), "amd64"),
+            ((Some("arm64"), Some("1.0.0"), "amd64", "2.0.0"), "arm64"),
+            ((None, Some("2.0.0-rc0"), "amd64", "2.0.0"), "amd64"),
```

Comment:
nit: SemVer (e.g. [in 2.0.0-rc.2][1]) and us (e.g. with 4.2.0-rc.0) put a dot after the `rc`.  Pre-releases [sort earlier][2], so doesn't really matter, but you may want `2.0.0-rc.0` here and below for consistency.

[1]: https://semver.org/spec/v2.0.0-rc.2.html
[2]: https://semver.org/spec/v2.0.0.html#spec-item-9